### ISSUE TITLE
Implement GTTK support

### ISSUE THREAD

**Issue by daemonspudguy** [shlomif/PySolFC#237]
GTTK is a project that allows Tkinter programs written in Python to use the GTK theme of a Linux system. This would make PySolFC fit in far better with other programs on the system.

https://github.com/TkinterEP/python-gttk

**Comment by MorsMortium** [comment]
Not sure if this is needed. I'm using gtkTtk https://github.com/Geballin/gtkTtk which themes every tk application.
![LLW9pWF](https://user-images.githubusercontent.com/44305161/162181293-433b3165-202c-4274-a0be-01dc4ea52983.png)

**Comment by daemonspudguy** [comment]
I can't get that to compile on Debian,    27 | #include <gdk-pixbuf-xlib/gdk-pixbuf-xlib.h>
@MorsMortium

**Comment by daemonspudguy** [comment]
I have gtkTtk installed and set as default, but PySol isn't using it.

**Comment by MorsMortium** [comment]
PySol has its own theme chooser, in Options/set theme. Though that would be an improvement, if it took the default theme, if it's set

**Comment by daemonspudguy** [comment]
It doesn't see gtkTtk, either.

**Comment by MorsMortium** [comment]
There's a single problem, that I had with it. But thats only if even the ttk theme chooser doesn't display the theme.
Like this:
![Screenshot_20220411_211313](https://user-images.githubusercontent.com/44305161/162812770-170088ff-b3ac-4981-a11a-bc97de0fe1a7.png)
That means it can't find your gtk theme, because it looks for the .gtkrc-2.0 file, in your home directory
Mine is like this, but with a dot in the beginning and without ".txt", github is weird.
[gtkrc-2.0.txt](https://github.com/shlomif/PySolFC/files/8467070/gtkrc-2.0.txt)
But if that's not your problem I'm not really sure, because this is the only problem i had with it.

**Comment by joeraz** [comment]
I tested this and was able to use the gtkTtk theme fine.  Installing gtkTtk added it to the Set Theme menu without issue.  I'll add looking into using the ttk theme chooser's default theme to my to do list, though I'd probably need to do a bit more research as to how that works before I'd be able to do that.

**Comment by daemonspudguy** [comment]
I think I know what I've been doing wrong. I've been installing PySolFC into a venv, as I couldn't figure out how else to install it.

**Comment by ssokolow** [comment]
It's a shame gtkTtk uses GTK+ 2.x or I'd include it in the Flatpak I'm working on to make a better first impression.

(Geeqie has taught me that Flatpak doesn't provide an auto-loadable theme package for the GTK 2.x version of Breeze, so GTK+ 2.x apps under Flatpak on KDE look terrible.)

I wonder how much work it'd be to resurrect the experimental GTK backend by porting it from PyGTK to PyGObject and then adding a copy of gtk-nocsd to the Flatpak image to ensure that it can find it if the host desktop specifies `LD_PRELOAD=libgtk3-nocsd.so.0`. (I imagine that'd also automatically give it native Wayland support since it shouldn't need to do the kind of X11-specific stuff my own ported project does.)

**Comment by hifron** [comment]
time to purge old themes and not let user think and if android needs something it would be exception if not found other solution... Time for GTK2(and maybe even 3) is up. If PyGTK make development a breeze, why not? There should be a lot of users of it...

**Comment by ssokolow** [comment]
> time to purge old themes and not let user think and if android needs something it would be exception if not found other solution... Time for GTK2(and maybe even 3) is up. If PyGTK make development a breeze, why not? There should be a lot of users of it...

Could you rephrase that? I understand your individual sentences, but I'm sure what stance they're all in support of.

**Comment by hifron** [comment]
https://github.com/shlomif/PySolFC/blob/master/docs/README.SOURCE

**A very experimental version for GTK+ 2 (using the pygnome and pygtk
  bindings) did exist as well. Due to pygnome and pygtk being obsoleted,
  a rewrite against GObject Introspection Python API would be needed
  to make the abandoned code work with GTK+ 3 or 4.**

```
PySol has been designed so that it can run under multiple UI toolkits -
  due to the dynamic nature of Python this can even be under control
  of a runtime option.

  The preferred toolkit is Tcl/Tk using the Tkinter bindings which
  ship with every Python installation.

  The Kivy UI adapts PySol for mobile devices and is mainly used for
  PySolFC's Android port. See README.kivy for more details.

  A very experimental version for GTK+ 2 (using the pygnome and pygtk
  bindings) did exist as well. Due to pygnome and pygtk being obsoleted,
  a rewrite against GObject Introspection Python API would be needed
  to make the abandoned code work with GTK+ 3 or 4.

  Another exciting idea is to use JPython to make PySol run under a
  Java VM using Swing as the toolkit.

  Because Tkinter is the "main" interface other toolkit layers have
  to emulate a limited subset of Tkinter's API. This should hopefully
  not prove too difficult in practice.
```

**Comment by ssokolow** [comment]
I'm aware of that. I'm asking what *you* meant by what you said. For example:

"time to purge old themes and not let user think and if android needs something it would be exception if not found other solution" is hard to make sense of.

For example, the "and not let users think" sounds like it's sarcastic, but I know some people whose first language isn't English who have used similar phrasings and intended them to mean "and protect users who just want to play the game from stepping on landmines" instead.

**Comment by hifron** [comment]
It shouldn't look like 8-bits and installed like it predecessor...

**Comment by ssokolow** [comment]
While I still don't feel confident I understand what thesis all of this is in support of, I think I understand enough to guess that this is off-topic and you should be opening a new issue that says something like "Delete GTK+ 2.x backend".

**Comment by hifron** [comment]
I would say find new programmers to make it happen their way and let Tk go. [Flutter](https://flutter.dev/games) missing but don't know how make it the Python way.

**Comment by ssokolow** [comment]
> I would say find new programmers to make it happen their way and let Tk go. [Flutter](https://flutter.dev/games) missing but don't know how make it the Python way.

Doesn't matter. It's still off-topic for this issue thread.

**Comment by MorsMortium** [comment]
> I tested this and was able to use the gtkTtk theme fine. Installing gtkTtk added it to the Set Theme menu without issue. I'll add looking into using the ttk theme chooser's default theme to my to do list, though I'd probably need to do a bit more research as to how that works before I'd be able to do that.

`~/.Xresources` has this line for me: `*TkTheme: gtkTtk`

**Comment by daemonspudguy** [comment]
Seeing this issue be revived 2 years after I wrote it with irrelevant nonsense outside the scope of the project was not something I expected to wake up to this morning.

**Comment by joeraz** [comment]
@MorsMortium - Was mostly thinking about how to pull the name of the theme from in the PySol code itself.

@hifron - Yes, make a new issue for this.  It would make it easier to track.

**Comment by MorsMortium** [comment]
@joeraz If you take a look at the actual theme chooser code, even that gets it from the file, I mentioned: https://github.com/Geballin/TTK-Theme-chooser/blob/master/ttk-theme-chooser.tcl
line 45
`proc actual_default_theme {} {`
